### DISCUSSION TITLE
Run Dawarich in a subdirectory

### DISCUSSION THREAD

**Discussion by Compizfox** [Freika/dawarich#330] [Q&A]
**Describe the bug**
Apologies in advance if this is documented somewhere and I missed it, but I couldn't find anything about this.

I am trying to run Dawarich with a reverse proxy in a subdirectory. In other words, I'm trying to reverse proxy `https://example.com/dawarich/` to the Dawarich app running in a Docker container at `http://localhost:3000`.

Unfortunately all the links in the frontend (including CSS and JS) are absolute links, so they point to `https://example.com/foobar` instead of `https://example.com/dawarich/foobar`, which results in it being completely broken.

Usually apps that use absolute URLs in this way provide a base URL variable that can be used for this instance, but I couldn't find that for Dawarich.

**Comment by Freika** [comment]
There is no such functionality in Dawarich at the moment. It's possible, but it's not a priority right now

**Comment by ignisf** [comment]
@Compizfox, did you try to set the `RAILS_RELATIVE_URL_ROOT` variable and see what happens?

**Reply by markerikson** [reply]
Does this actually work? I'm trying to get this exact scenario to cooperate.  I've confirmed that `RAILS_RELATIVE_URL_ROOT` is set in the container and is getting passed through to the Rails config as `Rails.application.config.relative_url_root`, but it doesn't seem to be having an effect on the generated URLs like `/users/sign_in`.

**Reply by lakemike** [reply]
I have the same question. I could see that RAILS_RELATIVE_URL_ROOT has an effect, but it does not seem to work overall. Is there any solution? 

Please allow me also to highlight the reason why this is an important feature from a privacy and security perspective: Essentially, the ability to use a "/prefix" for reverse proxy setups will give people the option to avoid using a full host name for the setup. This is good, because the host names can be discovered (see https://crt.sh). Once discovered they can be targeted by bots. Any host name like "dawarich.domain.tld" or something similar could be immediately targeted. Especially if there are gaps in authentication.

@Freika - Please consider removing the "answer" tag. It may still not work for anyone.

**Reply by gzin** [reply]
The same for me: there is the security thing. And beside that I am also having other apps on my webserver and public ip, the "/" isn't free any more. I don't want to announce an own subdomain and virtual server instance for that. Something like a configurable "/xyzdawarich" instead of "/" would be fine.

**Reply by christaikobo** [reply]
> I have the same question. I could see that RAILS_RELATIVE_URL_ROOT has an effect, but it does not seem to work overall. Is there any solution?
> 
> Please allow me also to highlight the reason why this is an important feature from a privacy and security perspective: Essentially, the ability to use a "/prefix" for reverse proxy setups will give people the option to avoid using a full host name for the setup. This is good, because the host names can be discovered (see https://crt.sh). Once discovered they can be targeted by bots. Any host name like "dawarich.domain.tld" or something similar could be immediately targeted. Especially if there are gaps in authentication.
> 
> @Freika - Please consider removing the "answer" tag. It may still not work for anyone.

What kind of "effect" did you see? I search the code base and RAILS_RELATIVE_URL_ROOT is not referenced anywhere which means setting this env probably does nothing right now.

**Reply by ignisf** [reply]
`RAILS_RELATIVE_ROOT_URL` is a framework-level switch. It's not expected to be mentioned in the code base. https://guides.rubyonrails.org/configuring.html#deploy-to-a-subdirectory-relative-url-root
